docs/contribute/plugin/develop_plugin_rustsdk.md

```diff
@@ -1,36 +1,79 @@
+---
+sidebar_position: 4
+---
+
 # Develop WasmEdge Plug-in in Rust SDK with witc
 
-Once you complete C++ plugin code, you can use witc[^1] to generate Rust Plugin SDK
+By developing a plugin, one can extend the functionality of WasmEdge and customize it to suit the specific needs. WasmEdge provides a Rust based API for registering extension modules and host functions.
+
+<!-- prettier-ignore -->
+:::note
+It is recommended that developers choose the WasmEdge [C API](develop_plugin_c.md) for plugin development because of the support, compatibility and flexibility provided by the WasmEdge runtime.
+:::
+
+## Set up the development environment
```

Comment:
I  don't think we need to repeat the development environment here. Just add a hyperlink here.

Reply:
Instruction for setting environment is different for each language. Then I am gonna keep the difference and for the rest will add hyperlink.